### ISSUE TITLE
Mergify doesn't always notice when my checks pass

### ISSUE THREAD

**Issue by bowlofeggs** [Mergifyio/mergify-engine#331]
It seems that #287 has returned. For example:

https://github.com/fedora-infra/bodhi/pull/2854

In that pull request, all checks passed, but the Mergify status shows quite a few as not passing:

![screenshot from 2019-01-07 12-14-24](https://user-images.githubusercontent.com/354506/50782394-da08e380-1275-11e9-97d9-380db65a20dd.png)

**Comment by sileht** [comment]
That's not the same issue, according our logs, it's because the organisation have reached the GitHub API rate-limit (5000 per hour).

**Comment by bowlofeggs** [comment]
Hi @sileht! Do you mean the fedora-infra organization has reached a rate limit, or do you mean that Mergifyio has?

**Comment by sileht** [comment]
Each Mergify installation have a rate-limit of 5000 request per hour. So it's Mergify GithubApp only for  the fedora-infra organization.

The number API calls per event to get a complete view of the pull request is around 10-15.

fedora-infra have 21 checks, so for each pull request create/update/rebase, this generates at least 210-315 Github API calls.

Also when when Github returns too many error 5XX, our retry mechanism was a bit too agressive, making the rate-limit reached earlier. I just readjusted it.

I keep this issue open and will track any further change that will reduce the number of API calls.

**Comment by bowlofeggs** [comment]
Ah interesting. We do have a few other projects in the fedora-infra org that have started using Mergify too, so that might explain some of the load as well.

**Comment by ryandawsonuk** [comment]
I've hit a similar situation. This PR is stuck:

https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/251#pullrequestreview-209078496

It says #commented-reviews-by=0 is not passing but I have approved it and resolved all comments. 

Almost immediately afterwards I resolved comments and approved https://github.com/Activiti/activiti-cloud-query-service/pull/194 and that went through fine. The only difference I can see in this particular case is that in the first case I approved and then resolved the comments (and then tried approving again but that didn't wake it up) whereas in the second I resolved the comments first and then approved.
